testservices/neutronservice/service_http_test.go

```diff
@@ -562,15 +580,33 @@ func (s *NeutronHTTPSuite) TestAddSecurityGroupRule(c *gc.C) {
 	c.Assert(*expected.Rule.PortRangeMin, gc.Equals, *rule1.PortRangeMin)
 	c.Assert(*expected.Rule.IPProtocol, gc.Equals, *rule1.IPProtocol)
 	c.Assert(expected.Rule.Direction, gc.Equals, rule1.Direction)
+	c.Assert(expected.Rule.RemoteIPPrefix, gc.Equals, rule1.RemoteIPPrefix)
+	// Attempt to create duplicate rule should fail
+	resp, err = s.jsonRequest("POST", neutron.ApiSecurityGroupRulesV2, req, nil)
+	c.Assert(resp.StatusCode, gc.Equals, http.StatusBadRequest)
 	defer s.service.removeSecurityGroupRule(rule1.Id)
-	req.Rule = riEgress
+	// Attempt to create rule with all fields but RemoteIPPrefix identical should pass
+	req.Rule = riIngress2
 	resp, err = s.jsonRequest("POST", neutron.ApiSecurityGroupRulesV2, req, nil)
 	c.Assert(err, gc.IsNil)
 	c.Assert(resp.StatusCode, gc.Equals, http.StatusCreated)
 	assertJSON(c, resp, &expected)
 	c.Assert(expected.Rule.Id, gc.Equals, rule2.Id)
 	c.Assert(expected.Rule.ParentGroupId, gc.Equals, rule2.ParentGroupId)
-	err = s.service.removeSecurityGroupRule(rule2.Id)
+	c.Assert(*expected.Rule.PortRangeMax, gc.Equals, *rule2.PortRangeMax)
+	c.Assert(*expected.Rule.PortRangeMin, gc.Equals, *rule2.PortRangeMin)
+	c.Assert(*expected.Rule.IPProtocol, gc.Equals, *rule2.IPProtocol)
+	c.Assert(expected.Rule.Direction, gc.Equals, rule2.Direction)
+	c.Assert(expected.Rule.RemoteIPPrefix, gc.Equals, rule2.RemoteIPPrefix)
+	defer s.service.removeSecurityGroupRule(rule2.Id)
```

Comment:
Thx! Will address these in the next PR  currently under review with a updated patch set before merge.